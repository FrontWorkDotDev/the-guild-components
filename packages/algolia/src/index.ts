/* eslint-disable no-console -- for debug */

import { createHash } from 'node:crypto';
import { existsSync } from 'node:fs';
import { readFile, writeFile } from 'node:fs/promises';
import path from 'node:path';
import algoliaSearch from 'algoliasearch';
import { XMLParser } from 'fast-xml-parser';
import GitHubSlugger from 'github-slugger';
import matter from 'gray-matter';
import sortBy from 'lodash.sortby';
import removeMarkdown from 'remove-markdown';
import { AlgoliaRecord, AlgoliaRecordSource, AlgoliaSearchItemTOC } from './types';

function extractToC(content: string): AlgoliaSearchItemTOC[] {
  const slugger = new GitHubSlugger();

  let isCodeBlock = false;
  let currentDepth = 0;
  let currentParent: AlgoliaSearchItemTOC | undefined;

  return content.split('\n').reduce<AlgoliaSearchItemTOC[]>((acc, value) => {
    if (value.match(/^```(.*)/)) {
      if (isCodeBlock) {
        isCodeBlock = false;
      } else {
        isCodeBlock = true;
        return acc;
      }
    } else if (isCodeBlock) {
      return acc;
    }

    const result = value.match(/(##+ )(.+)/);

    if (!result) return acc;

    const depth = result[1]?.length - 3;

    if (depth > 1) {
      return acc;
    }

    const heading = result[2]?.trim();

    const record: AlgoliaSearchItemTOC = {
      children: [],
      title: heading,
      anchor: slugger.slug(heading),
    };

    if (depth > 0) {
      currentParent?.children.push(record);
      if (depth > currentDepth) {
        currentParent = record;
      }
    } else {
      currentParent = record;
      acc.push(record);
    }

    currentDepth = depth;

    return acc;
  }, []);
}

const withoutTrailingSlashes = (str: string) => str.replace(/\/+$/, '');
const withTrailingSlash = (str: string) => `${withoutTrailingSlashes(str)}/`;

const contentForRecord = (content: string) => {
  let isCodeBlock = false;
  let isMeta = false;
  return removeMarkdown(
    content
      .split('\n')
      .map(line => {
        // remove code snippets
        if (line.match(/^```(.*)/)) {
          isCodeBlock = !isCodeBlock;
          return null;
        }
        if (isCodeBlock) {
          return null;
        }
        // remove metadata headers
        if (line.startsWith('---')) {
          isMeta = !isMeta;
          return null;
        }
        if (isMeta) {
          return null;
        }
        // remove titles
        if (line.startsWith('#')) {
          return null;
        }
        // remove `import` and `export`
        if (!isCodeBlock && (line.match(/^export(.*)/) || line.match(/^import(.*)/))) {
          return null;
        }
        return line;
      })
      .filter(line => line !== null)
      .join(' '),
  ).trim();
};

async function pluginsToAlgoliaRecords(
  // TODO: fix later
  plugins: any[],
  source: AlgoliaRecordSource,
  domain: string,
  objectsPrefix = new GitHubSlugger().slug(source),
): Promise<AlgoliaRecord[]> {
  const slugger = new GitHubSlugger();

  return plugins.map((plugin: any) => {
    const toc = extractToC(plugin.readme || '');
    return {
      objectID: slugger.slug(`${objectsPrefix}-${plugin.title}`),
      headings: toc.map(t => t.title),
      toc,
      content: contentForRecord(plugin.readme || ''),
      url: `${withTrailingSlash(domain)}plugins/${plugin.identifier}`,
      domain: withoutTrailingSlashes(domain),
      hierarchy: [source, 'Plugins'],
      source,
      title: plugin.title,
      type: 'Plugin',
    };
  });
}

interface IndexToAlgoliaNextraOptions {
  docsBaseDir: string;
  source: AlgoliaRecordSource;
  domain: string;
  sitemapXmlPath: string;
  objectsPrefix?: string;
}

async function getSitemapUrlsXml(sitemapXmlPath: string): Promise<string[]> {
  if (!sitemapXmlPath) {
    throw new Error('Sitemap path not provided');
  }
  if (!existsSync(sitemapXmlPath)) {
    throw new Error(`Sitemap at path "${sitemapXmlPath}" does not exist`);
  }
  const sitemap = new XMLParser().parse(await readFile(sitemapXmlPath, 'utf-8'));
  const sitemapUrls: { loc: string }[] = sitemap?.urlset?.url;
  if (!sitemapUrls?.length) {
    console.debug(sitemapXmlPath, JSON.stringify(sitemap, null, '  '));
    throw new Error('Sitemap urls not found at path urlset>url');
  }
  const urls = sitemapUrls.map(({ loc }) => loc).filter(Boolean);
  if (!urls?.length) {
    console.debug(sitemapXmlPath, JSON.stringify(sitemap, null, '  '));
    throw new Error('No sitemap urls found path urlset>url>loc');
  }
  return urls;
}

export async function nextraToAlgoliaRecords({
  docsBaseDir,
  source,
  domain,
  sitemapXmlPath,
  objectsPrefix = new GitHubSlugger().slug(source),
}: IndexToAlgoliaNextraOptions): Promise<AlgoliaRecord[]> {
  const sitemapUrls = await getSitemapUrlsXml(sitemapXmlPath);

  const objects: AlgoliaRecord[] = [];
  const slugger = new GitHubSlugger();

  for (const sitemapUrl of sitemapUrls) {
    const pageSlug = withoutTrailingSlashes(sitemapUrl.replace(withTrailingSlash(domain), ''));

    let pagePath = path.join(docsBaseDir, pageSlug);
    let pageContent: string;

    // since we strip extensions from urls, if the path exists - it's a directory, otherwise a markdown file
    const isDir = existsSync(pagePath);
    pagePath = `${pagePath}${isDir ? '/index' : ''}.md`;
    if (existsSync(pagePath)) {
      // .md
      pageContent = await readFile(pagePath, 'utf-8');
    } else if (((pagePath = `${pagePath}x`), existsSync(pagePath))) {
      // .mdx
      pageContent = await readFile(pagePath, 'utf-8');
    } else {
      throw new Error(`Page ${pagePath.replace('.mdx', '.{md,mdx}')} does not exist`);
    }

    // front-matter title or first appearing H1 heading content
    const {
      data: { title: matterTitle, type: matterType },
      content,
    } = matter(pageContent);
    let title = matterTitle;
    if (!title) {
      // title mustnt be on the first line
      for (const line of content.split('\n')) {
        if (line.startsWith('# ')) {
          title = line.replace('# ', '').trim();
          break;
        }
      }
    }
    if (!title) {
      throw new Error(`Title for page ${pagePath} not found`);
    }

    const hierarchy = pageSlug.split('/');
    if (!isDir) {
      // if the page is an actual file, then the directory (for the hierarchy) is one up
      hierarchy.pop();
    }

    const filenameWithoutExt = path.basename(pagePath, path.extname(pagePath));

    const toc = extractToC(content);

    objects.push({
      objectID: slugger.slug(
        `${objectsPrefix}-${[
          ...hierarchy,
          // TODO: should we remove the `index` file from the object ID?
          filenameWithoutExt.replaceAll('.', '_'),
        ].join('-')}`,
      ),
      headings: toc.map(t => t.title),
      toc,
      content: contentForRecord(content),
      url: sitemapUrl,
      domain: withoutTrailingSlashes(domain),
      hierarchy,
      source,
      title,
      type: matterType || 'Documentation',
    });
  }

  return objects;
}

export type { AlgoliaRecord, AlgoliaRecordSource, AlgoliaSearchItemTOC };

interface IndexToAlgoliaOptions {
  // TODO: fix later
  plugins?: any[];
  nextra?: IndexToAlgoliaNextraOptions;
  source: AlgoliaRecordSource;
  domain: string;
  sitemapXmlPath: string;
  lockfilePath: string;
  dryMode?: boolean;
  postProcessor?: (objects: AlgoliaRecord[]) => AlgoliaRecord[];
}

export async function indexToAlgolia({
  plugins = [],
  source,
  domain,
  sitemapXmlPath,
  nextra,
  postProcessor = value => value,
  // TODO: add `force` flag
  dryMode = true,
  lockfilePath,
}: IndexToAlgoliaOptions) {
  const objects = postProcessor([
    ...(await pluginsToAlgoliaRecords(plugins, source, domain)),
    ...(nextra
      ? await nextraToAlgoliaRecords({
          docsBaseDir: nextra.docsBaseDir,
          source,
          domain,
          sitemapXmlPath,
        })
      : []),
  ]);

  const recordsAsString = JSON.stringify(
    sortBy(objects, 'objectID'),
    (key, value) => (key === 'content' ? createHash('md5').update(value).digest('hex') : value),
    2,
  );

  const lockFileExists = existsSync(lockfilePath);
  const lockfileContent = JSON.stringify(
    // save space but still keep track of content changes
    sortBy(lockFileExists ? JSON.parse(await readFile(lockfilePath, 'utf8')) : [], 'objectID'),
    null,
    2,
  );

  if (dryMode) {
    console.log(`${lockfilePath} updated!`);
    await writeFile(lockfilePath, recordsAsString);
    return;
  }
  if (!lockFileExists || recordsAsString !== lockfileContent) {
    if (
      !process.env.ALGOLIA_APP_ID ||
      !process.env.ALGOLIA_ADMIN_API_KEY ||
      !process.env.ALGOLIA_INDEX_NAME
    ) {
      console.error('Some Algolia environment variables are missing!');
      return;
    }
    if (lockFileExists) {
      console.log('changes detected, updating Algolia index!');
    } else {
      console.log('no lockfile detected, push all records');
    }

    const client = algoliaSearch(process.env.ALGOLIA_APP_ID, process.env.ALGOLIA_ADMIN_API_KEY);
    const index = client.initIndex(process.env.ALGOLIA_INDEX_NAME);
    index
      .deleteBy({ filters: `source: "${source}"` })
      .then(() => index.saveObjects(objects))
      .then(console.log)
      .catch(console.error);

    await writeFile(lockfilePath, recordsAsString);
  }
}
