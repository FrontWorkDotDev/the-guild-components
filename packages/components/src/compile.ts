export * from 'nextra/compile';
export { remarkLinkRewrite } from 'nextra/mdx-plugins';
export { buildDynamicMDX, buildDynamicMeta } from 'nextra/remote';
export { fetchFilePathsFromGitHub } from 'nextra/fetch-filepaths-from-github';
