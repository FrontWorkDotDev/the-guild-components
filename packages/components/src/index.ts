export { default as Giscus } from '@giscus/react';
export {
  Bleed,
  Collapse,
  Navbar,
  NotFoundPage,
  useConfig,
  useTheme,
  useThemeConfig,
  type DocsThemeConfig,
} from 'nextra-theme-docs';
export { createCatchAllMeta } from 'nextra/catch-all';
export {
  Callout,
  Cards,
  Code,
  FileTree,
  Mermaid,
  RemoteContent,
  Steps,
  Tabs,
  Table,
  Td,
  Th,
  Tr,
} from 'nextra/components';
export { useData, useMounted } from 'nextra/hooks';
export { useMDXComponents } from 'nextra/mdx';
export * from './components';
export { defineConfig } from './define-config';
export { fetchPackageInfo } from './npm';
export { PRODUCTS } from './products';
export * from './types/components';
export * from './logos';

declare module 'react' {
  interface CSSProperties {
    [key: `--${string}`]: string | number | undefined;
  }
}
