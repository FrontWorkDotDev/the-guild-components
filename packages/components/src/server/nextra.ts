export { fetchFilePathsFromGitHub } from 'nextra/fetch-filepaths-from-github';
export { getPageMap } from 'nextra/page-map';
export { compileMdx } from 'nextra/compile'
