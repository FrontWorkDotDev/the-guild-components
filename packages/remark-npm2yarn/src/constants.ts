export const PACKAGE_MANAGERS = ['npm', 'pnpm', 'yarn'] as const;

export type PackageManager = (typeof PACKAGE_MANAGERS)[number];

export const KEY_VALUE_REGEX = /\w+="(.*?)"/g;

export const CHARS_REGEX = /\/(.*?)\/((#[a-z])|([\d-,])+)?/g;

export const LINES_REGEX = /\{(.*?)}/g;

export function cleanMeta(meta: string): string {
  return meta
    .replaceAll(KEY_VALUE_REGEX, '')
    .replaceAll(CHARS_REGEX, '')
    .replaceAll(LINES_REGEX, '');
}
