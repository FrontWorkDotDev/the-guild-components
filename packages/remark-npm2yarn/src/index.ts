export { remarkNpm2Yarn } from './plugin';
