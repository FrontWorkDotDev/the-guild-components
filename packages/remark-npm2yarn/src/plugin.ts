import { Code, Root } from 'mdast';
import convert from 'npm-to-yarn';
import { Plugin } from 'unified';
import { visit } from 'unist-util-visit';

const META_PLACEHOLDER = 'npm2yarn';
const TABS_PACKAGE_NAME = 'nextra-theme-docs';
const PACKAGE_MANAGERS = ['pnpm', 'yarn', 'npm'] as const;

type PackageManager = (typeof PACKAGE_MANAGERS)[number];

function getTabAST(node: Code, packageManager: PackageManager) {
  return {
    type: 'mdxJsxFlowElement',
    name: 'Tab',
    children: [
      {
        type: node.type,
        lang: node.lang,
        meta: node.meta?.replace(META_PLACEHOLDER, ''),
        value: convert(node.value, packageManager),
      },
    ],
  };
}

export const remarkNpm2Yarn: Plugin<[], Root> = () => (ast, _file, done) => {
  visit(ast, 'code', (node: Code, index, parent) => {
    const hasNpm2YarnMeta = node.meta?.includes(META_PLACEHOLDER);

    if (!hasNpm2YarnMeta) return;
    console.log('here', hasNpm2YarnMeta, index);

    // Replace current node with Tabs/Tab
    parent!.children[index!] = {
      type: 'mdxJsxFlowElement',
      name: 'Tabs',
      attributes: [
        {
          type: 'mdxJsxAttribute',
          name: 'items',
          value: {
            type: 'mdxJsxAttributeValueExpression',
            data: {
              estree: {
                type: 'Program',
                sourceType: 'module',
                body: [
                  {
                    type: 'ExpressionStatement',
                    expression: {
                      type: 'ArrayExpression',
                      elements: [
                        { type: 'Literal', value: 'pnpm' },
                        { type: 'Literal', value: 'yarn' },
                        { type: 'Literal', value: 'npm' },
                      ],
                    },
                  },
                ],
              },
            },
          },
        },
      ],
      children: [getTabAST(node, 'pnpm'), getTabAST(node, 'yarn'), getTabAST(node, 'npm')],
    } as any;

    // Add import statement at top of file
    ast.children.unshift({
      type: 'mdxjsEsm',
      data: {
        estree: {
          type: 'Program',
          sourceType: 'module',
          body: [
            {
              type: 'ImportDeclaration',
              source: { type: 'Literal', value: TABS_PACKAGE_NAME },
              specifiers: [
                {
                  type: 'ImportSpecifier',
                  imported: { type: 'Identifier', name: 'Tabs' },
                  local: { type: 'Identifier', name: 'Tabs' },
                },
                {
                  type: 'ImportSpecifier',
                  imported: { type: 'Identifier', name: 'Tab' },
                  local: { type: 'Identifier', name: 'Tab' },
                },
              ],
            },
          ],
        },
      },
    } as any);
  });

  done();
};
