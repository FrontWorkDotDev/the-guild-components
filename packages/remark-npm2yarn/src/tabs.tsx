import { Children, ReactElement, ReactNode, useCallback, useEffect, useState } from 'react';
import { Tabs as NextraTabs, Tab } from 'nextra-theme-docs';
import { PackageManager } from './constants';

const STORAGE_KEY = 'packageManager';

export function Tabs({
  children,
  packageManagers,
}: {
  children: ReactNode;
  packageManagers: PackageManager[];
}): ReactElement {
  const [selectedIndex, setSelectedIndex] = useState(0);

  useEffect(() => {
    function fn(event: StorageEvent) {
      if (event.key === STORAGE_KEY) {
        setSelectedIndex(packageManagers.indexOf(event.newValue as PackageManager));
      }
    }

    const index = packageManagers.indexOf(localStorage.getItem(STORAGE_KEY) as PackageManager);
    setSelectedIndex(index === -1 ? 0 : index);

    window.addEventListener('storage', fn);
    return () => {
      window.removeEventListener('storage', fn);
    };
    // eslint-disable-next-line react-hooks/exhaustive-deps -- only on mount
  }, []);

  const handleChange = useCallback((index: number) => {
    localStorage.setItem(STORAGE_KEY, packageManagers[index]);
    // the storage event only get picked up (by the listener) if the localStorage was changed in
    // another browser's tab/window (of the same app), but not within the context of the current tab.
    window.dispatchEvent(
      new StorageEvent('storage', {
        key: STORAGE_KEY,
        newValue: packageManagers[index],
      }),
    );
    // eslint-disable-next-line react-hooks/exhaustive-deps -- only on mount
  }, []);

  return (
    <NextraTabs selectedIndex={selectedIndex} onChange={handleChange} items={packageManagers}>
      {Children.map(children, (child, index) => (
        <Tab key={packageManagers[index]}>{child}</Tab>
      ))}
    </NextraTabs>
  );
}
