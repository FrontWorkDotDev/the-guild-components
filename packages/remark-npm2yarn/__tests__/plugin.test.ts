import { compileMdx } from 'nextra/compile';
import { remarkNpm2Yarn } from '../src';
import {
  CHARS_REGEX,
  cleanMetadataParam,
  KEY_VALUE_REGEX,
  LINES_REGEX,
  META_PLACEHOLDER,
} from '../src/constants';

describe('remark-npm2yarn', () => {
  it('should convert to tabs', async () => {
    const mdx = await compileMdx(
      `
\`\`\`sh filename="Terminal" npm2yarn
npm i -D @graphql-eslint/eslint-plugin
\`\`\`
    `,
      {
        mdxOptions: {
          remarkPlugins: [
            [
              remarkNpm2Yarn,
              {
                packageName: 'nextra/components',
                tabNamesProp: 'items',
                storageKey: 'testKey',
              },
            ],
          ],
        },
      },
    );
    expect(mdx.result).toMatchSnapshot();
  });
});

describe('clean up meta', () => {
  const META = [
    'baz1="foo npm2yarn"',
    '/npm2yarn/',
    META_PLACEHOLDER,
    '{npm2yarn}',
    'bar2="foo npm2yarn"',
    '/npm2yarn/3-5,6',
    'foo',
    '{npm2yarn}',
    '/npm2yarn/#v',
  ];

  const finalResult = '  npm2yarn    foo  ';

  it('should clean by regex', () => {
    const withoutKeyValues = META.join(' ').replaceAll(KEY_VALUE_REGEX, '');
    expect(withoutKeyValues).toBe(
      ' /npm2yarn/ npm2yarn {npm2yarn}  /npm2yarn/3-5,6 foo {npm2yarn} /npm2yarn/#v',
    );

    const withoutChars = withoutKeyValues.replaceAll(CHARS_REGEX, '');
    expect(withoutChars).toBe('  npm2yarn {npm2yarn}   foo {npm2yarn} ');

    const withoutLines = withoutChars.replaceAll(LINES_REGEX, '');
    expect(withoutLines).toBe(finalResult);
  });

  it('should clean by cleanMeta()', () => {
    expect(cleanMetadataParam(META.join(' '), META_PLACEHOLDER)).toBe(
      META.map(str => (str === META_PLACEHOLDER ? '' : str)).join(' '),
    );
  });
});
