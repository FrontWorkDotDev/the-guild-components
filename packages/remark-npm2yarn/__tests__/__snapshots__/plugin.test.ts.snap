// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`remark-npm2yarn > should convert to tabs 1`] = `
""use strict";
const {jsx: _jsx, jsxs: _jsxs} = arguments[0];
const title = "";
const metadata = {};
function useTOC(props) {
  return [];
}
function _createMdxContent(props) {
  const _components = {
    code: "code",
    pre: "pre",
    span: "span",
    ...props.components
  }, {$Tabs} = _components;
  if (!$Tabs) _missingMdxReference("$Tabs", true);
  if (!$Tabs.Tab) _missingMdxReference("$Tabs.Tab", true);
  return _jsxs($Tabs, {
    items: ["npm", "pnpm", "yarn", "bun"],
    storageKey: "testKey",
    children: [_jsx($Tabs.Tab, {
      children: _jsx(_components.pre, {
        tabIndex: "0",
        "data-language": "sh",
        "data-word-wrap": "",
        "data-filename": "Terminal",
        "data-pagefind-ignore": "all",
        children: _jsx(_components.code, {
          children: _jsxs(_components.span, {
            children: [_jsx(_components.span, {
              style: {
                "--shiki-light": "#6F42C1",
                "--shiki-dark": "#B392F0"
              },
              children: "npm"
            }), _jsx(_components.span, {
              style: {
                "--shiki-light": "#032F62",
                "--shiki-dark": "#9ECBFF"
              },
              children: " i"
            }), _jsx(_components.span, {
              style: {
                "--shiki-light": "#005CC5",
                "--shiki-dark": "#79B8FF"
              },
              children: " -D"
            }), _jsx(_components.span, {
              style: {
                "--shiki-light": "#032F62",
                "--shiki-dark": "#9ECBFF"
              },
              children: " @graphql-eslint/eslint-plugin"
            })]
          })
        })
      })
    }), _jsx($Tabs.Tab, {
      children: _jsx(_components.pre, {
        tabIndex: "0",
        "data-language": "sh",
        "data-word-wrap": "",
        "data-filename": "Terminal",
        "data-pagefind-ignore": "all",
        children: _jsx(_components.code, {
          children: _jsxs(_components.span, {
            children: [_jsx(_components.span, {
              style: {
                "--shiki-light": "#6F42C1",
                "--shiki-dark": "#B392F0"
              },
              children: "pnpm"
            }), _jsx(_components.span, {
              style: {
                "--shiki-light": "#032F62",
                "--shiki-dark": "#9ECBFF"
              },
              children: " add"
            }), _jsx(_components.span, {
              style: {
                "--shiki-light": "#005CC5",
                "--shiki-dark": "#79B8FF"
              },
              children: " -D"
            }), _jsx(_components.span, {
              style: {
                "--shiki-light": "#032F62",
                "--shiki-dark": "#9ECBFF"
              },
              children: " @graphql-eslint/eslint-plugin"
            })]
          })
        })
      })
    }), _jsx($Tabs.Tab, {
      children: _jsx(_components.pre, {
        tabIndex: "0",
        "data-language": "sh",
        "data-word-wrap": "",
        "data-filename": "Terminal",
        "data-pagefind-ignore": "all",
        children: _jsx(_components.code, {
          children: _jsxs(_components.span, {
            children: [_jsx(_components.span, {
              style: {
                "--shiki-light": "#6F42C1",
                "--shiki-dark": "#B392F0"
              },
              children: "yarn"
            }), _jsx(_components.span, {
              style: {
                "--shiki-light": "#032F62",
                "--shiki-dark": "#9ECBFF"
              },
              children: " add"
            }), _jsx(_components.span, {
              style: {
                "--shiki-light": "#005CC5",
                "--shiki-dark": "#79B8FF"
              },
              children: " --dev"
            }), _jsx(_components.span, {
              style: {
                "--shiki-light": "#032F62",
                "--shiki-dark": "#9ECBFF"
              },
              children: " @graphql-eslint/eslint-plugin"
            })]
          })
        })
      })
    }), _jsx($Tabs.Tab, {
      children: _jsx(_components.pre, {
        tabIndex: "0",
        "data-language": "sh",
        "data-word-wrap": "",
        "data-filename": "Terminal",
        "data-pagefind-ignore": "all",
        children: _jsx(_components.code, {
          children: _jsxs(_components.span, {
            children: [_jsx(_components.span, {
              style: {
                "--shiki-light": "#6F42C1",
                "--shiki-dark": "#B392F0"
              },
              children: "bun"
            }), _jsx(_components.span, {
              style: {
                "--shiki-light": "#032F62",
                "--shiki-dark": "#9ECBFF"
              },
              children: " add"
            }), _jsx(_components.span, {
              style: {
                "--shiki-light": "#005CC5",
                "--shiki-dark": "#79B8FF"
              },
              children: " --dev"
            }), _jsx(_components.span, {
              style: {
                "--shiki-light": "#032F62",
                "--shiki-dark": "#9ECBFF"
              },
              children: " @graphql-eslint/eslint-plugin"
            })]
          })
        })
      })
    })]
  });
}
return {
  title,
  metadata,
  useTOC,
  default: _createMdxContent
};
function _missingMdxReference(id, component) {
  throw new Error("Expected " + (component ? "component" : "object") + " \`" + id + "\` to be defined: you likely forgot to import, pass, or provide it.");
}
"
`;
